packages/pages/src/dev/server/ssr/findMatchingModule.ts

```diff
@@ -30,3 +35,86 @@ export const findModuleInternal = async (
 
   return null;
 };
+
+export const loadModuleInternalFromFilepath = async (
+  devserver: ViteDevServer,
+  moduleFilePath: string
+): Promise<ModuleInternal> => {
+  const moduleModule = await loadViteModule<ModuleModule>(
+    devserver,
+    moduleFilePath
+  );
+
+  return convertModuleToModuleInternal(moduleFilePath, moduleModule);
+};
+
+export const getPostCssPathFromModuleName = async (
+  moduleName: string | undefined,
+  projectStructure: ProjectStructure
+): Promise<string> => {
+  if (moduleName == undefined) {
+    return "";
+  }
+  const moduleFilePaths =
+    getModuleFilepathsFromProjectStructure(projectStructure);
+  const project = createTsMorphProject();
+  let targetModuleFilePath = "";
+  for (const moduleFilePath of moduleFilePaths) {
+    project.addSourceFileAtPath(moduleFilePath);
+    const sourceFile = project.getSourceFile(moduleFilePath);
+    if (sourceFile !== undefined) {
+      const exportedDeclarations = sourceFile.getExportedDeclarations();
+      for (const [name, declarations] of exportedDeclarations) {
+        if (name == "config") {
+          declarations.forEach((declaration) => {
+            for (const child of declaration.getChildren()) {
+              const text = child.getText();
+              if (text.includes("name:")) {
+                const parsedName = text
+                  .substring(
+                    text.indexOf('"') + 1,
+                    text.indexOf('"', text.indexOf('"') + 1)
+                  )
+                  .toLowerCase();
+                if (parsedName == moduleName.toLowerCase()) {
+                  targetModuleFilePath = moduleFilePath;
+                  break;
+                }
+              }
+            }
+          });
+        }
+      }
+    }
+  }
+
+  // Use file path that matches given name if none match the config.name, if possible
+  if (targetModuleFilePath == "") {
+    for (const moduleFilePath of moduleFilePaths) {
+      const fileName = moduleFilePath.split("/").slice(-1)[0];
+      if (
+        fileName.substring(0, fileName.indexOf(".")).toLowerCase() ==
```

Comment:
We don't care about filename, we care about folder name

Reply:
In Diana's instructions it says that the filename and folder name have to match. I can dig a bit more and see if that's actually enforced or just recommended.

I'll change this to check the folder name instead though.